### ISSUE TITLE
Can't remove image created by Buildah

### ISSUE THREAD

**Issue by TomSweeneyRedHat** [projectatomic/libpod#174]
**Is this a BUG REPORT or FEATURE REQUEST?**:
kind bug

**Description**
An image created by Buildah is listed by Podman but can not be removed by Podman.  It can be removed by Buildah and then it is no longer shown by Podman.

**Steps to reproduce the issue:**

1. buildah from alpine
2. podman images
3. podman rmi -f {imageid}

**Describe the results you received:**
 ```
# buildah from alpine
# podman images
REPOSITORY                 TAG      IMAGE ID       CREATED       SIZE
docker.io/library/alpine   latest   b908b128a5f6   4 weeks ago   4.14MB
# podman rmi -f b908b128a5f6
failed to remove image: Image used by 2b1024cf1d55d555af8ce15ed009fd9fb3c691c362d7a1c7c90107a0cf84371e: image is in use by a container
# podman ps --all
# buildah images
IMAGE ID             IMAGE NAME                                               CREATED AT             SIZE
b908b128a5f6         docker.io/library/alpine:latest                          Dec 1, 2017 18:48      3.954 MB
# buildah rmi -f b908b128a5f6
ERRO[0000] reference "[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.override_kernel_check=true]docker.io/library/b908b128a5f6:latest" does not resolve to an image ID 
b908b128a5f6172525f2fd0fed13ee5812ee168f45b623fc9955033568888153
# podman images

// Note the error from Buildah is a known error being tracked in https://github.com/projectatomic/buildah/issues/190
```

**Describe the results you expected:**
The image would be removed via Podman or the image in the list should somehow be marked as a Buildah image or otherwise non-deletable by Podman.  It's more confusing as Podman says there are no containers when queried.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
# podman version
Version:       0.1
Go Version:    go1.9.2
Git Commit:    "ef4035e208f1431745cd044adb4b789fab3a98b6"
Built:         Tue Jan  2 08:27:22 2018
OS/Arch:       linux/amd64
# podman -v
podman version 0.1
# buildah -v
buildah version 0.9 (image-spec 1.0.0, runtime-spec 1.0.0)
# cat /etc/*release
Fedora release 27 (Twenty Seven)
NAME=Fedora
VERSION="27 (Server Edition)"
ID=fedora
VERSION_ID=27
PRETTY_NAME="Fedora 27 (Server Edition)"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:fedoraproject:fedora:27"
HOME_URL="https://fedoraproject.org/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=27
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=27
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Server Edition"
VARIANT_ID=server
Fedora release 27 (Twenty Seven)
Fedora release 27 (Twenty Seven)

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

On a VM running Fedora 27 updated as of 1/2/2018

**Comment by mheon** [comment]
Buildah has mounted the image, but since it doesn't share state with libpod, we don't know that it is mounted and do not attempt to unmount it before removing. It is probably safe to make force-removing an image do an unconditional unmount after unmounting all containers using the image.

**Comment by TomSweeneyRedHat** [comment]
Fixed with #522
